### DISCUSSION TITLE
Dynamically change a modals action buttons / Change a ViewAction to a EditAction from within the view form

### DISCUSSION THREAD

**Discussion by BradnDodd** [filamentphp/filament#14153] [Help]
### Package

Actions

### Package Version

3.2.103

### How can we help you?

What I'm trying to achieve is a view button against a table record that opens a modal with a disabled form with an edit button within the modal itself that once click will reload the form with the fields enabled and a save changes button to appear in the same modal footer actions as the normal buttons. So far I have the form enabling working but I cannot get the save changes button to work properly

This is my action; 
```
EditAction::make('view')
	->label('View')
	->slideOver()
	->disabledForm()
	->form(self::form($modelLabel))
	->modalSubmitAction(false)
	->modalCancelAction(fn (StaticAction $action) => $action->label('Close'))
	->extraModalFooterActions([
		Action::make('reload')
			->label('Edit')
			->action(function (Action $action) use ($modelLabel) {
				$action->getTable()
					->getAction('view')
					->disabledForm(false)
					->form(self::form($modelLabel, showEditReason: true))
					->render();
			})
	]),
``` 

I have tried setting the `->modalSubmitAction()` back to a StaticAction from within `->extraModalFooterActions()` but this does not work. I suspect that is because I am not calling render on the correct part. 

Any ideas on how I can either achieve the modal re-rendering using my current Action setup or suggest a different way of achieving the same thing, thanks in advance

**Comment by e2d2-dev** [comment]
Hi,

```
ViewAction::make()
	->slideOver()
	->form(self::form($modelLabel))
	->modalSubmitAction(false)
	->extraModalFooterActions([
		EditAction::make()
			->record(fn($record) => $record)
	]),
```

should do the trick
